aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -296,12 +296,8 @@ Tensor& add_out_dense_sparse_cuda(Tensor& r_, const Tensor& dense, const SparseT
   Tensor dense_buffer = dense.to(commonDtype);
   Tensor values = sparse._values().to(commonDtype);
 
-  if (is_same_tensor(r, dense_buffer)) {
```

Comment:
You need to keep this if statement!
Only resize/copy if r is not dense_buffer.

Reply:
Oh...yea